app/src/main/java/org/mozilla/fenix/settings/DeleteBrowsingDataController.kt

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings
+
+import android.content.Context
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.withContext
+import mozilla.components.concept.engine.Engine
+import mozilla.components.feature.tab.collections.TabCollection
+import org.mozilla.fenix.FeatureFlags
+import org.mozilla.fenix.ext.components
+import kotlin.coroutines.CoroutineContext
+
+interface DeleteBrowsingDataController {
+    suspend fun deleteTabs()
+    suspend fun deleteBrowsingData()
+    suspend fun deleteCollections(collections: List<TabCollection>)
+    suspend fun deleteCookies()
+    suspend fun deleteCachedFiles()
+    suspend fun deleteSitePermissions()
+}
+
+class DefaultDeleteBrowsingDataController(
+    val context: Context,
+    val coroutineContext: CoroutineContext = Dispatchers.Main
+) : DeleteBrowsingDataController {
+    override suspend fun deleteTabs() {
+        withContext(coroutineContext) {
+            context.components.useCases.tabsUseCases.removeAllTabs.invoke()
+        }
+    }
+
+    override suspend fun deleteBrowsingData() {
+        withContext(coroutineContext) {
+            if (FeatureFlags.granularDataDeletion) {
+                context.components.core.engine.clearData(Engine.BrowsingData.select(Engine.BrowsingData.DOM_STORAGES))
```

Comment:
Where did you get that DOM storages should be part of this one? I'm not sure I saw this in my mocks for delete on quit. We should just make sure what we're clearing is consistent

Reply:
It seems like if we want everything selected to delete everything, we don't want to miss something. Where do DOM storages belong on this UI? With cached files and images for recovering disk space?

Reply:
I agree with you that it looks like `DOM_STORAGES` and `AUTH_SESSIONS` now belong to none of these categories. We should probably verify with product and UX where these should be cleared and then keep it consistent between hte two features?

Reply:
I believe AUTH_SESSIONS really belongs in its own category, but DOM_STORAGES are a type of history or cache, depending upon how you look at it. The reason I included it here is because the delete cache checkbox mentions that its intent is to reduce storage usage. Deleting history and DOM storage offers more of a privacy improvement than storage savings.

Reply:
I move AUTH_SESSIONS to the cookies category based on feedback from the GV team. I think DOM_STORAGES belongs here based upon the descriptions we're offering to users.